modules/misc/gtk/lib.nix

```diff
@@ -72,7 +73,9 @@ in
     // optionalAttrs (cursorTheme != null) { "gtk-cursor-theme-name" = cursorTheme.name; }
     // optionalAttrs (cursorTheme != null && cursorTheme.size != null) {
       "gtk-cursor-theme-size" = cursorTheme.size;
-    };
+    }
+    // optionalAttrs (colorScheme == "dark") { "gtk-application-prefer-dark-theme" = true; }
+    // optionalAttrs (colorScheme == "light") { "gtk-application-prefer-light-theme" = true; };
```

Comment:
Isn't it just a single option with true/false? https://docs.gtk.org/gtk3/property.Settings.gtk-application-prefer-dark-theme.html

Reply:
Indeed, my bad, I naively thought the color-scheme mechanism worked the same for `application-prefer`.
Fixed now, let me now if my solution works for you or you think it's too convoluted.